LiteCore/Query/SQLiteDataFile+Indexes.cc

```diff
@@ -99,6 +105,20 @@ namespace litecore {
                     case IndexSpec::kFullText:
                     case IndexSpec::kVector:
                         same = schemaExistsWithSQL(indexTableName, "table", indexTableName, indexSQL);
+                        if ( spec.type == IndexSpec::kFullText ) {
+                            if ( same ) {
+                                auto whatA = spec.what(), whatB = existingSpec->what();
+                                if ( !whatA ? whatB != nullptr : !whatB ) same = false;
```

Comment:
Could what be practically NULL? I understand that it's a required field.

Reply:
I think the purpose here is to check the sameness. NULL should be avoided somewhere else.
I tried it, empty expression will cause crash before coming to here.
Do platforms check it?

Reply:
If LiteCore returns the error for the empty expression, the platform will return or throw an exception to the user for that error.